### PR TITLE
support knative 0.19+

### LINKED ISSUE #1333: Knative serving 0.19+ breaks the top level KFServing virtual service
/kind bug

**What steps did you take and what happened:**
I am trying to access the sklearn-iris example using the xip.io service running on minikube with kfserving. I am able to successfully use the xip.io to access the both of http://sklearn-iris-predictor-canary.myns.192.168.49.2.xip.io and http://sklearn-iris-predictor-default.myns.192.168.49.2.xip.io , but I am unable to access the http://sklearn-iris.myns.192.168.49.2.xip.io.

```
curl -v -d @iris-input.json http://sklearn-iris.myns.192.168.49.2.xip.io:30194/v1/models/sklearn-iris:predict 
*   Trying 192.168.49.2...
* TCP_NODELAY set
* Connected to sklearn-iris.myns.192.168.49.2.xip.io (192.168.49.2) port 30194 (#0)
> POST /v1/models/sklearn-iris:predict HTTP/1.1
> Host: sklearn-iris.myns.192.168.49.2.xip.io:30194
> User-Agent: curl/7.58.0
> Accept: */*
> Content-Length: 76
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 76 out of 76 bytes
< HTTP/1.1 503 Service Unavailable
< content-length: 91
< content-type: text/plain
< date: Fri, 29 Jan 2021 17:07:08 GMT
< server: istio-envoy
< 
* Connection #0 to host sklearn-iris.myns.192.168.49.2.xip.io left intact
upstream connect error or disconnect/reset before headers. reset reason: connection failure
```

**What did you expect to happen:**
Top level KFS virtual service should also work.

**Anything else you would like to add:**
Knative 0.20 deprecated the local cluster gateway and uses the service `knative-local-gateway` which listens on port 8081.

```
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: knative-local-gateway
  namespace: knative-serving
  labels:
    serving.knative.dev/release: "v0.20.0"
    networking.knative.dev/ingress-provider: istio
spec:
  selector:
    istio: ingressgateway
  servers:
    - port:
        number: 8081
        name: http
        protocol: HTTP
      hosts:
        - "*"
---
apiVersion: v1
kind: Service
metadata:
  name: knative-local-gateway
  namespace: istio-system
  labels:
    serving.knative.dev/release: "v0.20.0"
    networking.knative.dev/ingress-provider: istio
spec:
  type: ClusterIP
  selector:
    istio: ingressgateway
  ports:
    - name: http2
      port: 80
      targetPort: 8081
```    

Currently KFS uses [a constant](https://github.com/kubeflow/kfserving/blob/master/pkg/constants/constants.go#L121) for local gateway service name which needs to be made configurable. 

**Environment:**

- Istio Version:
- Knative Version: 0.20
- KFServing Version: 0.4.1
- Kubeflow version:
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by theofpa**
**What this PR does / why we need it**:
Adds support for knative 0.19+ (and 0.20).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1333

**Special notes for your reviewer**:
Work in progress, we should replace the `KnativeLocalGateway` constant with an option from the inferenceservice ConfigMap so users of old knative can still use kfserving 0.5.

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Support Knative 0.20 and Istio 1.7.1
```

### CODE REVIEW

**Comment by yuzisun** [comment]
Thanks @theofpa ! 

Let's also bump up the knative version [in e2e test](https://github.com/kubeflow/kfserving/blob/master/test/scripts/run-e2e-tests.sh#L27) ?

File: test/scripts/run-e2e-tests.sh [hunk 1]
```diff
@@ -150,11 +117,11 @@ sed -i -e "s/latest/${PULL_BASE_SHA}/g" config/overlays/test/manager_image_patch
 make deploy-ci
 
 echo "Waiting for KFServing started ..."
-kubectl wait --for=condition=ready pod -l control-plane=kfserving-controller-manager -n kfserving-system
+kubectl wait --for=condition=Ready pods --all --timeout=180s -n kfserving-system
 
 echo "Creating a namespace kfserving-ci-test ..."
 kubectl create namespace kfserving-ci-e2e-test
-
+kubectl label namespace kfserving-ci-e2e-test istio-injection=enabled
```
Comment (yuzisun) [inline test/scripts/run-e2e-tests.sh:124]: Do we need to inject the sidecar?

Comment (theofpa) [reply test/scripts/run-e2e-tests.sh]: removed in 9276b12

**Comment by yuzisun** [comment]
@theofpa Are you going to change to read from the configuration?

**Comment by theofpa** [comment]
link to #1363 about making cluster-local-gateway configurable

**Comment by yuzisun** [comment]
@theofpa seems an error during install
```
nstalling knative serving ...
customresourcedefinition.apiextensions.k8s.io/knativeeventings.operator.knative.dev created
customresourcedefinition.apiextensions.k8s.io/knativeservings.operator.knative.dev created
configmap/config-logging created
configmap/config-observability created
deployment.apps/knative-operator created
clusterrole.rbac.authorization.k8s.io/knative-serving-operator-aggregated created
clusterrole.rbac.authorization.k8s.io/knative-serving-operator created
clusterrole.rbac.authorization.k8s.io/knative-eventing-operator-aggregated created
clusterrole.rbac.authorization.k8s.io/knative-eventing-operator created
clusterrolebinding.rbac.authorization.k8s.io/knative-serving-operator created
clusterrolebinding.rbac.authorization.k8s.io/knative-serving-operator-aggregated created
clusterrolebinding.rbac.authorization.k8s.io/knative-eventing-operator created
clusterrolebinding.rbac.authorization.k8s.io/knative-eventing-operator-aggregated created
serviceaccount/knative-operator created
namespace/knative-serving created
knativeserving.operator.knative.dev/knative-serving created
Waiting for knative started ...
✔ Installation completeerror: no matching resources found
```

**Comment by theofpa** [comment]
> @theofpa seems an error during install
> 
> ```
> nstalling knative serving ...
> customresourcedefinition.apiextensions.k8s.io/knativeeventings.operator.knative.dev created
> customresourcedefinition.apiextensions.k8s.io/knativeservings.operator.knative.dev created
> configmap/config-logging created
> configmap/config-observability created
> deployment.apps/knative-operator created
> clusterrole.rbac.authorization.k8s.io/knative-serving-operator-aggregated created
> clusterrole.rbac.authorization.k8s.io/knative-serving-operator created
> clusterrole.rbac.authorization.k8s.io/knative-eventing-operator-aggregated created
> clusterrole.rbac.authorization.k8s.io/knative-eventing-operator created
> clusterrolebinding.rbac.authorization.k8s.io/knative-serving-operator created
> clusterrolebinding.rbac.authorization.k8s.io/knative-serving-operator-aggregated created
> clusterrolebinding.rbac.authorization.k8s.io/knative-eventing-operator created
> clusterrolebinding.rbac.authorization.k8s.io/knative-eventing-operator-aggregated created
> serviceaccount/knative-operator created
> namespace/knative-serving created
> knativeserving.operator.knative.dev/knative-serving created
> Waiting for knative started ...
> ✔ Installation completeerror: no matching resources found
> ```

yes, it seems that the operator takes some time before it starts creating the resources and the kubectl wait fails to find any pods in the new namespace.

**Comment by yuzisun** [comment]
@theofpa seems it always fails on the logger test, is it possible that the gateway change breaks the logger?

**Comment by theofpa** [comment]
> @theofpa seems it always fails on the logger test, is it possible that the gateway change breaks the logger?

There was a json syntax error in the configmap and the inferenceservice was not becoming ready, let's give it one more try!

**Comment by yuzisun** [comment]
> > @theofpa seems it always fails on the logger test, is it possible that the gateway change breaks the logger?
> 
> There was a json syntax error in the configmap and the inferenceservice was not becoming ready, let's give it one more try!

still the logger test :(

**Comment by theofpa** [comment]
triggering the test again to check the logs on the ephemeral cluster
/retest

**Comment by theofpa** [comment]
So, it failed again but now I have some information from the ephemeral cluster:

Both `message-dumper` and `idvc-logger` isvcs start:
```
NAMESPACE               NAME                     URL                                                       READY     PREV   LATEST   PREVROLLEDOUTREVISION   LATESTREADYREVISION                              AGE
kfserving-ci-e2e-test   isvc-logger              http://isvc-logger.kfserving-ci-e2e-test.example.com      True             100                              isvc-logger-predictor-default-00002              70s
kfserving-ci-e2e-test   message-dumper           http://message-dumper.kfserving-ci-e2e-test.example.com   True             100                              message-dumper-predictor-default-00002           93s
```

And their pods readiness checks are ok:

```
NAME                                                              READY   STATUS        RESTARTS   AGE
isvc-logger-predictor-default-00001-deployment-75844b99d5-tw9l9   3/3     Running       0          65s
isvc-logger-predictor-default-00002-deployment-788dccb745-fz28c   3/3     Running       0          65s
message-dumper-predictor-default-00001-deployment-777f5767bcj6n   2/2     Running       0          63s
message-dumper-predictor-default-00002-deployment-78cc45d6gc5jc   2/2     Running       0          63s
```

Their ksvcs are ok:

```
NAME                               URL                                                                         LATESTCREATED                            LATESTREADY                              READY   REASON
isvc-logger-predictor-default      http://isvc-logger-predictor-default.kfserving-ci-e2e-test.example.com      isvc-logger-predictor-default-00002      isvc-logger-predictor-default-00002      True    
message-dumper-predictor-default   http://message-dumper-predictor-default.kfserving-ci-e2e-test.example.com   message-dumper-predictor-default-00002   message-dumper-predictor-default-00002   True  
```

The agent container of isvc-logger though, reports that it can't connect to the `message-dumper.kfserving-ci-e2e-test.svc.cluster.local ` to send the cloud-event:

```
$ kubectl logs pod/isvc-logger-predictor-default-00002-deployment-788dccb745-fz28c -n kfserving-ci-e2e-test -c agent
{"level":"info","ts":1613085461.9505332,"logger":"fallback-logger","caller":"logging/config.go:78","msg":"Fetch GitHub commit ID from kodata failed","error":"\"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1613085461.950582,"logger":"fallback-logger","caller":"agent/main.go:249","msg":"Starting the log dispatcher"}
{"level":"info","ts":1613085461.9505956,"logger":"fallback-logger","caller":"logger/dispatcher.go:31","msg":"Starting worker 1"}
{"level":"info","ts":1613085461.9506137,"logger":"fallback-logger","caller":"logger/dispatcher.go:31","msg":"Starting worker 2"}
{"level":"info","ts":1613085461.9506297,"logger":"fallback-logger","caller":"logger/dispatcher.go:31","msg":"Starting worker 3"}
{"level":"info","ts":1613085461.9506385,"logger":"fallback-logger","caller":"logger/dispatcher.go:31","msg":"Starting worker 4"}
{"level":"info","ts":1613085461.9506433,"logger":"fallback-logger","caller":"logger/dispatcher.go:31","msg":"Starting worker 5"}
aggressive probe error:  dial tcp 127.0.0.1:8080: connect: connection refused
aggressive probe error:  dial tcp 127.0.0.1:8080: connect: connection refused
...
aggressive probe error:  dial tcp 127.0.0.1:8080: connect: connection refused
aggressive probe error:  dial tcp 127.0.0.1:8080: connect: connection refused
{"level":"info","ts":1613085513.8220925,"logger":"fallback-logger","caller":"logger/worker.go:132","msg":"Received work request 4, url: http://message-dumper.kfserving-ci-e2e-test.svc.cluster.local, requestId: 53081997-4976-4eba-aa1d-afa5261f2401"}
{"level":"info","ts":1613085513.8267047,"logger":"fallback-logger","caller":"logger/worker.go:132","msg":"Received work request 1, url: http://message-dumper.kfserving-ci-e2e-test.svc.cluster.local, requestId: 53081997-4976-4eba-aa1d-afa5261f2401"}
{"level":"error","ts":1613085513.8676097,"logger":"fallback-logger","caller":"logger/worker.go:135","msg":"while sending event: Post \"http://message-dumper.kfserving-ci-e2e-test.svc.cluster.local\": dial tcp: lookup message-dumper.kfserving-ci-e2e-test.svc.cluster.local on 10.100.0.10:53: no such hostFailed to send cloud event, url: %shttp://message-dumper.kfserving-ci-e2e-test.svc.cluster.local","stacktrace":"github.com/kubeflow/kfserving/pkg/logger.(*Worker).Start.func1\n\t/go/src/github.com/kubeflow/kfserving/pkg/logger/worker.go:135"}
{"level":"error","ts":1613085513.8677151,"logger":"fallback-logger","caller":"logger/worker.go:135","msg":"while sending event: Post \"http://message-dumper.kfserving-ci-e2e-test.svc.cluster.local\": dial tcp: lookup message-dumper.kfserving-ci-e2e-test.svc.cluster.local on 10.100.0.10:53: no such hostFailed to send cloud event, url: %shttp://message-dumper.kfserving-ci-e2e-test.svc.cluster.local","stacktrace":"github.com/kubeflow/kfserving/pkg/logger.(*Worker).Start.func1\n\t/go/src/github.com/kubeflow/kfserving/pkg/logger/worker.go:135"}
```

The vs for host `message-dumper.kfserving-ci-e2e-test.svc.cluster.local` is using the gateway `knative-local-gateway.knative-serving`:

```
NAME                                         GATEWAYS                                                                          HOSTS                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       AGE
message-dumper                               [knative-local-gateway.knative-serving knative-ingress-gateway.knative-serving]   [message-dumper.kfserving-ci-e2e-test.svc.cluster.local message-dumper.kfserving-ci-e2e-test.example.com]                                                                                                                                                                                                                                                                                                                                                                                                                                                                   11m
message-dumper-predictor-default-ingress     [knative-serving/knative-ingress-gateway knative-serving/knative-local-gateway]   [latest-message-dumper-predictor-default.kfserving-ci-e2e-test latest-message-dumper-predictor-default.kfserving-ci-e2e-test.example.com latest-message-dumper-predictor-default.kfserving-ci-e2e-test.svc latest-message-dumper-predictor-default.kfserving-ci-e2e-test.svc.cluster.local message-dumper-predictor-default.kfserving-ci-e2e-test message-dumper-predictor-default.kfserving-ci-e2e-test.example.com message-dumper-predictor-default.kfserving-ci-e2e-test.svc message-dumper-predictor-default.kfserving-ci-e2e-test.svc.cluster.local]                   11m
message-dumper-predictor-default-mesh        [mesh]                                                                            [latest-message-dumper-predictor-default.kfserving-ci-e2e-test latest-message-dumper-predictor-default.kfserving-ci-e2e-test.svc latest-message-dumper-predictor-default.kfserving-ci-e2e-test.svc.cluster.local message-dumper-predictor-default.kfserving-ci-e2e-test message-dumper-predictor-default.kfserving-ci-e2e-test.svc message-dumper-predictor-default.kfserving-ci-e2e-test.svc.cluster.local]                                                                                                                                                                11m
```

The gateways have been created as expected in the knative namespace:

```
$ kubectl get gateway -A
NAMESPACE         NAME                      AGE
knative-serving   knative-ingress-gateway   18m
knative-serving   knative-local-gateway     18m
```

The gateway service has also been created as expected in the istio namespace:

```
$ kubectl get svc -n istio-system
NAME                    TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                         AGE
istiod                  ClusterIP   10.100.115.182   <none>        15010/TCP,15012/TCP,443/TCP,15014/TCP,853/TCP   19m
knative-local-gateway   ClusterIP   10.100.228.17    <none>        80/TCP                                          18m
```

The vs of the message-dumper points to the right gateway service and right port:

```
$ kubectl get vs -n kfserving-ci-e2e-test message-dumper -o yaml
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  creationTimestamp: "2021-02-11T23:17:23Z"
  generation: 1
  name: message-dumper
  namespace: kfserving-ci-e2e-test
  ownerReferences:
  - apiVersion: serving.kubeflow.org/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: InferenceService
    name: message-dumper
    uid: 0f83b8b7-1b99-47b6-a90e-f6b1ec250e0e
  resourceVersion: "9976"
  selfLink: /apis/networking.istio.io/v1beta1/namespaces/kfserving-ci-e2e-test/virtualservices/message-dumper
  uid: 8d2eeb39-4754-46c5-adee-9254d3d96ebe
spec:
  gateways:
  - knative-local-gateway.knative-serving
  - knative-ingress-gateway.knative-serving
  hosts:
  - message-dumper.kfserving-ci-e2e-test.svc.cluster.local
  - message-dumper.kfserving-ci-e2e-test.example.com
  http:
  - match:
    - authority:
        regex: ^message-dumper\.kfserving-ci-e2e-test(\.svc(\.cluster\.local)?)?(?::\d{1,5})?$
      gateways:
      - knative-local-gateway.knative-serving
    - authority:
        regex: ^message-dumper\.kfserving-ci-e2e-test\.example\.com(?::\d{1,5})?$
      gateways:
      - knative-ingress-gateway.knative-serving
    route:
    - destination:
        host: knative-local-gateway.istio-system.svc.cluster.local
        port:
          number: 80
      headers:
        request:
          set:
            Host: message-dumper-predictor-default.kfserving-ci-e2e-test.svc.cluster.local
      weight: 100
```

**Comment by theofpa** [comment]
This `cluster-local-gateway` in the svcs looks like a leftover, it should be `knative-local-gateway`, I'll try to find it and send a new commit.

```
$ kubectl get svc -n kfserving-ci-e2e-test
NAME                                                 TYPE           CLUSTER-IP       EXTERNAL-IP                                            PORT(S)                             AGE
isvc-logger                                          ExternalName   <none>           cluster-local-gateway.istio-system.svc.cluster.local   <none>                              9m7s
isvc-logger-predictor-default                        ExternalName   <none>           knative-local-gateway.istio-system.svc.cluster.local   80/TCP                              9m8s
isvc-logger-predictor-default-00001                  ClusterIP      10.100.175.15    <none>                                                 80/TCP                              9m49s
isvc-logger-predictor-default-00001-private          ClusterIP      10.100.134.49    <none>                                                 80/TCP,9090/TCP,9091/TCP,8022/TCP   9m49s
isvc-logger-predictor-default-00002                  ClusterIP      10.100.90.172    <none>                                                 80/TCP                              9m49s
isvc-logger-predictor-default-00002-private          ClusterIP      10.100.172.128   <none>                                                 80/TCP,9090/TCP,9091/TCP,8022/TCP   9m49s
latest-isvc-logger-predictor-default                 ExternalName   <none>           knative-local-gateway.istio-system.svc.cluster.local   80/TCP                              9m8s
latest-isvc-triton-predictor-default                 ExternalName   <none>           knative-local-gateway.istio-system.svc.cluster.local   80/TCP                              2s
latest-message-dumper-predictor-default              ExternalName   <none>           knative-local-gateway.istio-system.svc.cluster.local   80/TCP                              10m
message-dumper                                       ExternalName   <none>           cluster-local-gateway.istio-system.svc.cluster.local   <none>                              10m
message-dumper-predictor-default                     ExternalName   <none>           knative-local-gateway.istio-system.svc.cluster.local   80/TCP                              10m
message-dumper-predictor-default-00001               ClusterIP      10.100.78.122    <none>                                                 80/TCP                              10m
message-dumper-predictor-default-00001-private       ClusterIP      10.100.63.122    <none>                                                 80/TCP,9090/TCP,9091/TCP,8022/TCP   10m
message-dumper-predictor-default-00002               ClusterIP      10.100.31.120    <none>                                                 80/TCP                              10m
message-dumper-predictor-default-00002-private       ClusterIP      10.100.41.151    <none>                                                 80/TCP,9090/TCP,9091/TCP,8022/TCP   10m
```

File: docs/samples/v1beta1/transformer/torchserve_image_transformer/input.json [hunk 2]
```diff
@@ -1,9 +0,0 @@
-
```
Comment (yuzisun) [inline docs/samples/v1beta1/transformer/torchserve_image_transformer/input.json:1]: revert this back?

File: test/scripts/run-e2e-tests.sh [hunk 3]
```diff
@@ -170,5 +138,5 @@ popd
 
 echo "Starting E2E functional tests ..."
 pushd test/e2e >/dev/null
-  pytest -n 3 --ignore=credentials/test_set_creds.py
-popd
+  pytest -n 4 --ignore=credentials/test_set_creds.py
+popd
```
Comment (yuzisun) [inline test/scripts/run-e2e-tests.sh:142]: nit: newline

**Comment by yuzisun** [comment]
@theofpa Can you help update the [developer guide](https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md#install-knative-on-a-kubernetes-cluster) as after this change it would require Knative 0.19 for development by default, otherwise configmap needs to be updated to the `cluster-local-gateway`

File: test/scripts/run-e2e-tests.sh (hunk 3 above)
Comment (theofpa) [reply test/scripts/run-e2e-tests.sh]: addressed in 81ca10c

File: docs/samples/v1beta1/transformer/torchserve_image_transformer/input.json (hunk 2 above)
Comment (theofpa) [reply docs/samples/v1beta1/transformer/torchserve_image_transformer/input.json]: oops, addressed in 81ca10c

File: install/v0.5.0/kfserving.yaml [hunk 4]
```diff
@@ -332,8 +332,8 @@ data:
     {
         "ingressGateway" : "knative-ingress-gateway.knative-serving",
         "ingressService" : "istio-ingressgateway.istio-system.svc.cluster.local",
-        "localGateway" : "cluster-local-gateway.knative-serving",
-        "localGatewayService" : "cluster-local-gateway.istio-system.svc.cluster.local"
+        "localGateway" : "knative-local-gateway",
```
Comment (yuzisun) [inline install/v0.5.0/kfserving.yaml:335]:
```suggestion
        "localGateway" : "cluster-local-gateway",
```

File: docs/DEVELOPER_GUIDE.md [hunk 5]
```diff
@@ -53,11 +53,15 @@ You must install these tools:
...
-
-* You need follow the instructions on [Updating your install to use cluster local gateway](https://knative.dev/docs/install/installing-istio/#updating-your-install-to-use-cluster-local-gateway) to add cluster local gateway to your cluster to serve cluster-internal traffic for transformer and explainer use cases.
+* If you already have `Istio` or `Knative` (e.g. from a Kubeflow install) then you don't need to install them explictly, as long as version dependencies are satisfied.
+  * With Kubeflow v0.7, KNative 0.8 and Istio 1.1.6 are installed by default as part of the Kubeflow installation.
+  * From Kubeflow 1.0 onwards, KNative 0.11.1 and Istio 1.1.6 are installed by default.
```
Comment (yuzisun) [inline docs/DEVELOPER_GUIDE.md:60]: This section can be removed as Istio 1.1 is EOL

File: install/v0.5.0/kfserving.yaml [hunk 6]
```diff
@@ -332,8 +332,8 @@ data:
...
+        "localGatewayService" : "knative-local-gateway.istio-system.svc.cluster.local"
```
Comment (yuzisun) [inline install/v0.5.0/kfserving.yaml:336]:
```suggestion
        "localGatewayService" : "cluster-local-gateway.istio-system.svc.cluster.local"
```

Comment (yuzisun) [reply install/v0.5.0/kfserving.yaml:336]: Let's not modify the released yaml, we can change the default in next version.

**Comment by theofpa** [comment]
The ephemeral cluster has [4 nodes](https://github.com/kubeflow/kfserving/blob/1311a7c65d20aece7b946683fd02957f488ee891/test/scripts/create-cluster.sh#L35) with `m5.xlarge` instances and can handle higher concurrency, so setting it to 4.

File: install/v0.5.0/kfserving.yaml [hunk 7]
```diff
@@ -332,7 +332,7 @@ data:
     {
         "ingressGateway" : "knative-ingress-gateway.knative-serving",
         "ingressService" : "istio-ingressgateway.istio-system.svc.cluster.local",
-        "localGateway" : "cluster-local-gateway.knative-serving",
+        "localGateway" : "cluster-local-gateway",
```
Comment (yuzisun) [inline install/v0.5.0/kfserving.yaml:335]: do not modify the release yaml

**Comment by yuzisun** [comment]
@theofpa One last comment otherwise good to merge

File: install/v0.5.0/kfserving.yaml (hunk 7 above)
Comment (theofpa) [reply install/v0.5.0/kfserving.yaml]: Rolled back!

**Comment by yuzisun** [comment]
thanks @theofpa ! This is awesome work!

/lgtm
/approve

**Pull request merged**